### PR TITLE
Allow users to request a specific e2e test to retest

### PR DESCRIPTION
**Pull request by fejta**
Look for the following:

```
gke test  # gke
gke e2e test # gke
e2e test # gce&gke&node
test  # gce&gke&node&unit&verify
```

---
This change is [Reviewable](https://reviewable.io/reviews/kubernetes/test-infra/441)

### CODE REVIEW

File: jenkins/job-configs/kubernetes-jenkins-pull/kubernetes-pull.yaml [hunk 1]
```diff
@@ -414,7 +414,7 @@
             * [Internal Jenkins Results](${{JOB_URL}}/${{BUILD_NUMBER}})
 
             Please reference the [list of currently known flakes](https://github.com/kubernetes/kubernetes/issues?q=is:issue+label:kind/flake+is:open) when examining this failure. If you request a re-test, you must reference the issue describing the flake.
-          trigger-phrase: 'e2e\s+test'
+          trigger-phrase: 'gce e2e test'
```
Comment (ixdy) [inline jenkins/job-configs/kubernetes-jenkins-pull/kubernetes-pull.yaml]: commenting "federation gce e2e test" will also trigger this.

and I predict a lot of frustrated developers wondering why "@k8s-bot e2e test this" isn't working. we probably need to widely announce and educate.

Comment (fejta) [inline jenkins/job-configs/kubernetes-jenkins-pull/kubernetes-pull.yaml]: Yes, I'll send an announcement after merging this.

I don't see what you're seeing with the top comment. Can you double-check this claim and/or explain further?

**Comment by rmmh** [comment]
So if you have "gce e2e test" it will trigger kubernetes-e2e-gce? Does it have to be on a line by itself?

Other people have been talking about using / commands to interact with the various repo bots -- so `/test` or `/test gce e2e` would fit better in that framework. Let's pick the right one before reeducating people.

**Comment by fejta** [comment]
Ryan please see where `trigger-phrase` is used. 

I think how we request testing is orthogonal to being able to distinguish between gke and gce testing. RIght now this is impossible. How about if I just disambiguate gke and gce and leave the gce command the same for now?

File: jenkins/job-configs/kubernetes-jenkins-pull/kubernetes-pull.yaml (hunk 1 above)
Comment (ixdy) [inline jenkins/job-configs/kubernetes-jenkins-pull/kubernetes-pull.yaml]: you're right, I forgot we add `@k8s-bot ` to the beginning of the trigger phrase.

**Comment by spxtr** [comment]
`@k8s-bot (gce )?(e2e )?test this` and `@k8s-bot (gke )?(e2e )?test this`?

**Comment by fejta** [comment]
Okay all previous commands still work.

Now `test this` will test everything, and I can request all e2e or a specific node|gke|gce e2e test too.

**Comment by ixdy** [comment]
I'm not sure if this is really better... https://xkcd.com/1171/

File: jenkins/job-configs/kubernetes-jenkins-pull/kubernetes-pull.yaml [hunk 2]
```diff
@@ -676,7 +676,7 @@
             owner: 'pwittroc@google.com'
             max-total: 10
             skip-if-no-test-files:   # old branches may not produce JUnit
-            trigger-phrase: 'node e2e test'
+            trigger-phrase: '(node)? (e2e )?test'
```
Comment (spxtr) [inline jenkins/job-configs/kubernetes-jenkins-pull/kubernetes-pull.yaml]: Move the space inside the parens.

**Comment by spxtr** [comment]
I think it's better.

File: jenkins/job-configs/kubernetes-jenkins-pull/kubernetes-pull.yaml (hunk 2 above)
Comment (fejta) [inline jenkins/job-configs/kubernetes-jenkins-pull/kubernetes-pull.yaml]: Done

**Comment by rmmh** [comment]
LGTM

**Pull request merged**